RockLib.Collections/NamedCollection.cs

```diff
@@ -6,203 +6,208 @@
+      /// </param>
+      /// <exception cref="ArgumentNullException">Thrown if <paramref name="values"/> or <paramref name="getName"/> are <c>null</c></exception>
+      public NamedCollection(IEnumerable<T> values, Func<T, string> getName,
+          IEqualityComparer<string>? stringComparer = null, string? defaultName = _defaultDefaultName, bool strict = true)
+      {
+         if (values is null)
+            throw new ArgumentNullException(nameof(values));
+         if (getName is null)
+            throw new ArgumentNullException(nameof(getName));
+
+         StringComparer = stringComparer ?? OrdinalIgnoreCase;
+         DefaultName = !string.IsNullOrEmpty(defaultName) ? defaultName! : _defaultDefaultName;
+
+         var valuesByName = new Dictionary<string, T>(StringComparer);
+         var hasDefaultValue = false;
+
+         foreach (var value in values)
+         {
+            var name = getName(value);
+            if (IsDefaultName(name))
             {
-                var name = getName(value);
-                if (IsDefaultName(name))
-                {
-                    if (strict && hasDefaultValue)
-                        throw new ArgumentException("Cannot have more than one default value.", nameof(values));
-                    hasDefaultValue = true;
-                    DefaultValue = value;
-                }
-                else
-                {
-                    if (strict && valuesByName.ContainsKey(name))
-                        throw new ArgumentException($"Cannot have more than one value with the same name: {name}.", nameof(values));
-                    valuesByName[name] = value;
-                }
+               if (strict && hasDefaultValue)
+                  throw new ArgumentException("Cannot have more than one default value.", nameof(values));
+               hasDefaultValue = true;
+               DefaultValue = value;
             }
-
-            _dictionaryKeySelector = value =>
+            else
             {
-                var name = getName(value);
-                return IsDefaultName(name) ? DefaultName : name;
-            };
+               if (strict && valuesByName.ContainsKey(name))
+                  throw new ArgumentException($"Cannot have more than one value with the same name: {name}.", nameof(values));
+               valuesByName[name] = value;
+            }
+         }
+
+         _dictionaryKeySelector = value =>
+         {
+            var name = getName(value);
+            return IsDefaultName(name) ? DefaultName : name;
+         };
 
-            Count = valuesByName.Count + (hasDefaultValue ? 1 : 0);
+         Count = valuesByName.Count + (hasDefaultValue ? 1 : 0);
 
-            _valuesByName = valuesByName;
+         _valuesByName = valuesByName;
 
 #if NET451
             NamedValues = valuesByName.Values.ToList();
 #else
-            NamedValues = valuesByName.Values;
+         NamedValues = valuesByName.Values;
```

Comment:
I think we can get rid of the #if NET451 right?

Reply:
Oops, missed that, I'll get rid of it.